tests/sandbox/LegacyInputIterator.h

```diff
@@ -0,0 +1,90 @@
+// FIXME remove pragma after moving to cts
+#pragma once
+
+#include "common.h"
+
+#include "EqualityComparable.h"
+#include "LegacyIterator.h"
+
+template <typename It>
+void check_legacy_input_iterator_requirement(It valid_iterator,
+                                             const size_t size_of_container,
+                                             const std::string& type_name) {
+  if (!iterator_verification_common::check_preconditions(
+          valid_iterator, size_of_container, type_name)) {
+    return;
+  }
+
+  check_legacy_iterator_requirement(valid_iterator, size_of_container,
+                                    type_name);
+  check_equality_comparable_requirement(valid_iterator, type_name);
+
+  It j = valid_iterator;
+  It i = valid_iterator;
+
+  // As "Legacy Iterator" implements increment operator we can get two not
+  // equal iterators
+  i = ++i;
+  {
+    INFO(
+        "Two not equal iterators have to return the same value with NOT EQUAL "
+        "operator");
+    CHECK();
+    CHECK((!(i == j)) == (i != j));
+  }
+  {
+    INFO(
+        "Two not equal iterators have to return implicit convertble to "
+        "bool value with NOT EQUAL operator");
+    CHECK(std::is_convertible_v<decltype((i != j)), bool>);
+    CHECK(std::is_convertible_v<decltype((i == j)), bool>);
```

Comment:
```suggestion
    CHECK((std::is_convertible_v<decltype((i != j)), bool>));
    CHECK((std::is_convertible_v<decltype((i == j)), bool>));
```
Don't we need extra brackets here? `CHECK` is a macro after all, and C++ preprocessor handles every `comma` as a macro parameter separator.

Reply:
Makes sense. Add extra brackets for safety